src/anonymous-credentials/presentation-builder.ts

```diff
@@ -492,6 +560,94 @@ export class PresentationBuilder extends Versioned {
+        const messages = cred.serializeForSigning();
+        const encoder = schema.encoder;
+        const [names, encodedValues] = encoder.encodeMessageObject(messages);
+        let found = 0;
+        for (let j = 0; j < names.length; j++) {
+          if (attributeNames.has(names[j])) {
+            attributes.push(encodedValues[j]);
+            const witnessEq = new WitnessEqualityMetaStatement();
+            witnessEq.addWitnessRef(credIdx, j);
+            witnessEqs.push([witnessEq, attributes.length - 1]);
+            found++;
+          }
+        }
+        if (attributeNames.size !== found) {
+          throw new Error(
+            `Some of the attribute name for pseudonym were not found in the given messages object, ${
+              attributeNames.size - found
+            } extra names found`
+          );
+        }
+      }
+      const pseudonym = AttributeBoundPseudonym.new(
+        attributeBoundPseudonym.basesForAttributes,
+        attributes,
+        attributeBoundPseudonym.baseForSecretKey,
+        attributeBoundPseudonym.secretKey
+      );
+      const baseForSecretKey = attributeBoundPseudonym.baseForSecretKey;
+      const decodedBaseForSecretKey =
+        baseForSecretKey !== undefined ? PseudonymBases.decodeBaseForSecretKey(baseForSecretKey) : undefined;
+      const decodedBasesForAttributes = PseudonymBases.decodeBasesForAttributes(
+        attributeBoundPseudonym.basesForAttributes
+      );
+      const decodedPseudonym = pseudonym.base64;
+      presentedBoundedPseudonyms.set(decodedPseudonym, {
+        commitKey: {
+          basesForAttributes: decodedBasesForAttributes,
+          baseForSecretKey: decodedBaseForSecretKey
+        },
+        attributes: attributeBoundPseudonym.attributeNames
+      });
+      const statement = Statement.attributeBoundPseudonym(
+        pseudonym,
+        attributeBoundPseudonym.basesForAttributes,
+        attributeBoundPseudonym.baseForSecretKey
+      );
+      const sIdx = statements.add(statement);
+      for (const [witnessEq, attributeIndex] of witnessEqs) {
+        witnessEq.addWitnessRef(sIdx, attributeIndex);
+        metaStatements.addWitnessEquality(witnessEq);
+      }
+      const witness = Witness.attributeBoundPseudonym(attributes, attributeBoundPseudonym.secretKey);
+      witnesses.add(witness);
+    }
+    this.spec.boundedPseudonyms = presentedBoundedPseudonyms;
+
+    // Create statements and witnesses for each unboundedPseudonyms
+    const presentedUnboundedPseudonyms: Map<string, IPresentedUnboundedPseudonym> = new Map();
+    for (let i = 0; i < this.unboundedPseudonyms.length; i++) {
+      const unboundedPseudonym = this.unboundedPseudonyms[i];
+      const pseudonym = Pseudonym.new(unboundedPseudonym.baseForSecretKey, unboundedPseudonym.secretKey);
+      const decodedBaseForSecretKey = PseudonymBases.decodeBaseForSecretKey(unboundedPseudonym.baseForSecretKey);
+      const decodedPseudonym = PseudonymBases.decode(pseudonym.value);
```

Comment:
This is slightly misleading as `pseudonym` is not a base. Better to define `encode`, `decode` on  `Pseudonym`. Use them above as well instead of `pseudonym.base64`

Reply:
I find it weird to have `Pseudonym.encode(value)` and the `decode` equivalent when both `AttributeBoundPseudonym` and `Pseudonym` derive from `BytearrayWrapper`. The `encode` and `decode` static functions I use for AttributeBoundPseudonym are taken from Pseudonym in a very arbitrary way.
I'd find it more appropriate to use (static OR not) functions from `BytearrayWrapper`.
Though I think it is just a question of style, I'll do whatever you prefer.
I just implemented it in `Pseudonym` like you mentioned.